Compiling the C++ SDK on Windows fails due to usage of STR_EXPAND in firestore_version.cc

When building the Firestore C++ SDK (https://github.com/firebase/firebase-cpp-sdk) on Windows via cmake, the build fails with the following error:

```
external\src\firestore\Firestore\core\src\firestore_version.cc(34,45): warning C4003: not enough arguments for function-like macro invocation 'STR_EXPAND'
external\src\firestore\Firestore\core\src\firestore_version.cc(34,70): error C2059: syntax error: ';'
```

The following code is the culprit: https://github.com/firebase/firebase-ios-sdk/blob/44ab89ec8d9ad817bf6fded3144e829a32ef9f10/Firestore/core/src/firestore_version.cc#L31-L34

The issue is that `FIRFirestore_VERSION` is empty/unset which causes the `STR_EXPAND` to expand to invalid code. When building via cmake, `FIRFirestore_VERSION` is set here: https://github.com/firebase/firebase-ios-sdk/blob/44ab89ec8d9ad817bf6fded3144e829a32ef9f10/Firestore/core/CMakeLists.txt#L223-L228

The `podspec_version` function is defined here: https://github.com/firebase/firebase-ios-sdk/blob/44ab89ec8d9ad817bf6fded3144e829a32ef9f10/cmake/podspec_rules.cmake#L17-L26

Since the `podspec_version` function uses `sed`, which is not by default installed on Windows, then if `sed` is not present the command will fail and the variable will be unset or set to the empty string.

The fix is to modify this code to either (a) use a mechanism that is cross platform to extract the version from the podspec or (b) fail hard if the command fails.

--- a/cmake/podspec_rules.cmake
+++ b/cmake/podspec_rules.cmake
@@ -28,7 +28,7 @@ macro(podspec_version VARIABLE PODSPEC_FILE)
       FATAL_ERROR
       "Running the sed script ${PROJECT_SOURCE_DIR}/cmake/podspec_version.sed \
       on file ${PODSPEC_FILE} failed; ensure that the `sed` executable is \
-      installed and that its directoroy is present in the PATH environment \
+      installed and that its directory is present in the PATH environment \
       variable."
     )
   endif()
@@ -48,7 +48,7 @@ macro(firebase_version VARIABLE PODSPEC_FILE)
       FATAL_ERROR
       "Running the sed script ${PROJECT_SOURCE_DIR}/cmake/firebase_version.sed \
       on file ${PODSPEC_FILE} failed; ensure that the `sed` executable is \
-      installed and that its directoroy is present in the PATH environment \
+      installed and that its directory is present in the PATH environment \
       variable."
     )
   endif()

--- a/cmake/podspec_rules.cmake
+++ b/cmake/podspec_rules.cmake
@@ -23,6 +23,15 @@ macro(podspec_version VARIABLE PODSPEC_FILE)
     OUTPUT_VARIABLE ${VARIABLE}
     OUTPUT_STRIP_TRAILING_WHITESPACE
   )
+  if(${VARIABLE} STREQUAL "")
+    message(
+      FATAL_ERROR
+      "Running the sed script ${PROJECT_SOURCE_DIR}/cmake/podspec_version.sed \
+      on file ${PODSPEC_FILE} failed; ensure that the `sed` executable is \
+      installed and that its directoroy is present in the PATH environment \
+      variable."
+    )
+  endif()
 endmacro()
 
 macro(firebase_version VARIABLE PODSPEC_FILE)
@@ -34,6 +43,15 @@ macro(firebase_version VARIABLE PODSPEC_FILE)
     OUTPUT_VARIABLE ${VARIABLE}
     OUTPUT_STRIP_TRAILING_WHITESPACE
   )
+  if(${VARIABLE} STREQUAL "")
+    message(
+      FATAL_ERROR
+      "Running the sed script ${PROJECT_SOURCE_DIR}/cmake/firebase_version.sed \
+      on file ${PODSPEC_FILE} failed; ensure that the `sed` executable is \
+      installed and that its directoroy is present in the PATH environment \
+      variable."
+    )
+  endif()
 endmacro()
 
 function(podspec_prep_headers FRAMEWORK_NAME)

